### DISCUSSION TITLE
How to prevent auto-horizontal scroll when selecting a TreeViewItem

### DISCUSSION THREAD

**Discussion by cornerbowlsoftware** [AvaloniaUI/Avalonia#17020] [General]
I would like to prevent the TreeView from auto-horizontal scrolling when I click on an item that overflows horizontally.  I solved this in my WPF app by overriding OnRequestBringIntoView then manually calling ScrollViewer ScrollToLeftEnd which doesn't exist in Avalonia.  In my WPF app, this also forces me to handle the vertical scroll which I accomplish with several variables, TreeViewItem.Height, ScrollViewer ActualHeight, ScrollViewer.Bounds.Height, and SystemParameters.HorizontalScrollBarHeight.  Before I tackle the vertical scroll, I need to handle the horizontal scroll. Does anyone know how this can be accomplished?

**Comment by BreakByBrick** [comment]
Hello!

For horizontal scrolling I solved this by overriding the **OnRequestBringIntoView** method in my TreeNode.
Scrolling to the tree node is done by the value of RequestBringIntoViewEventArgs.TargetRect, I have set the Width value to zero.

```
protected override void OnRequestBringIntoView( RequestBringIntoViewEventArgs e )
{
	base.OnRequestBringIntoView( e );

	e.TargetRect = e.TargetRect.WithWidth( 0 );
}
```

You can disable scrolling entirely using the TreeView.AutoScrollToSelectedItem property. For example, you can override its default value to false, in your tree.

```
static TreeViewBase()
{
	AutoScrollToSelectedItemProperty.OverrideDefaultValue<TreeViewBase>( false );
}
```
